### ISSUE TITLE
Test loss = 1 / decoder giving back labels

### ISSUE THREAD

**Issue by jtcampbellsoup** [vrenkens/nabu#58]
Hi Vincent,

I've successfully trained the model using a different alphabet. I see during training time the loss on the dev set goes down from about 100 at the start to around 17 at the end. However, when I do the training/decoding steps, the testing prints that the loss I get is 1.0, which doesn't make sense. The decoder also creates the file of the decodings, but it doesn't list any of the phones, only the label of that testing example. Here is what the decoded file looks like:
![image](https://user-images.githubusercontent.com/49572572/58054979-34159180-7b11-11e9-8815-8a68fbe9819a.png)
Do you know why the error printed would be 1.0 and/or why it's not printing out any phones?

Here is an example of my train.txt file, which maps a label to it's phones
![image](https://user-images.githubusercontent.com/49572572/58054726-565adf80-7b10-11e9-9a06-fab5f1bc56f8.png)
and my train.scp file, which maps a label to it's audio file
![image](https://user-images.githubusercontent.com/49572572/58055244-fc5b1980-7b11-11e9-84dc-2d27880b0621.png)
Just putting that there to see if this format is correct. Really appreciate your help.

**Comment by vrenkens** [comment]
Hey, the thing that seems to be happening is that it is always outputting empty sequences, which causes a WER of 100% (or 1.0). I don't really know what causing this... Could you also provide the test files? Maybe something is wrong there.

**Comment by jtcampbellsoup** [comment]
Here is the model that was saved. All of the config files from the recipe are in there as well. I omitted the logdir folder to save space.
[DBLSTM_02.zip](https://github.com/vrenkens/nabu/files/3201802/DBLSTM_02.zip)

The test.scp and test.txt files look exactly the same as the pictures shown above, but with different labels mapping to different phones / audio files. Same with the dev files.

Some things to note:
1) I did a very quick training, only going through one epoch, because I just want a crude baseline model at the moment. This should hurt performance, but I don't think it explains the blank labels.
2) The decoder specified in model.cfg is "dnn_decoder" (this one does not exist,, I accidentally didn't change this before training) and it is "ctc_decoder" in both the test_evaluator.cfg and recognizer.cfg files. Don't think this explains it either, though, since I think it would have thrown an error in training when it didn't find the dnn_decoder.

**Comment by vrenkens** [comment]
It is possible that, because you did a very crude training the model will only output empty sequences. The thing is that the "blank" label in CTC is the most common label, so if the model has not trained sufficiently it might just use a guessing strategy. Since the blank label is the most common one the best guessing strategy is always outputting the blank label explaining the empty output sequences
